app/api/resources/user.py

```diff
@@ -248,16 +248,14 @@ def get(cls):
 @users_ns.route("register")
 class UserRegister(Resource):
     @classmethod
-    @users_ns.doc("create_user")
-    @users_ns.response(200, "%s" % messages.USER_WAS_CREATED_SUCCESSFULLY)
-    @users_ns.response(
-        400,
-        "%s\n%s"
-        % (
-            messages.USER_USES_A_USERNAME_THAT_ALREADY_EXISTS,
-            messages.USER_USES_AN_EMAIL_ID_THAT_ALREADY_EXISTS,
-        ),
-    )
+    @users_ns.doc('create_user')
+    @users_ns.response(200, '%s'%messages.USER_WAS_CREATED_SUCCESSFULLY)
+    @users_ns.response(400, '%s\n%s\n%s'%(
+        messages.USER_USES_A_USERNAME_THAT_ALREADY_EXISTS,
+        messages.USER_USES_AN_EMAIL_ID_THAT_ALREADY_EXISTS,
+        messages.REGISTRATION_FAILED
+    ))
+    @users_ns.response(500, '%s'%messages.REGISTRATION_FAILED)
```

Comment:
500 should not be Registration Failed, if we're returning 500, we must indicate that the user should contact the mentorship-backend team or raise an Issue. We need to be more verbose with **why** the registration failed, and help the user understand that it is not an incorrect operation on their end.

Reply:
@ramitsawhney27, for instead of 500, I'm thinking perhaps 407 Proxy Authentication Required is more suitable in this error type coz it is related to authentication of the user's email. Will this be more appropriate?